plugin/Makefile

```diff
@@ -74,15 +76,21 @@ TMP_LOC := /tmp/plugin
 
 # default target
 all: info plugin push
+shared-all: shared-info shared-plugin shared-push
```

Comment:
I still suggest postpone this to a little bit later stage.
These changes will have to change the CI and testings too.
We don't want to disturb the testing code while the new shared plugin is still under development.
So, please think this new plugin as an experimental feature, which won't be a default choice for now :)